yoti-sdk-api/src/main/java/com/yoti/api/client/shareurl/policy/WantedAttribute.java

```diff
@@ -135,10 +152,35 @@ public Builder withConstraint(Constraint constraint) {
             return this;
         }
 
+        public Builder withAlternativeNames(Set<String> alternativeNames) {
+            this.alternativeNames = Collections.unmodifiableSet(alternativeNames);
```

Comment:
I realise this was being done, but using `Collections.unmodifiableSet` here doesn't make sense, it _doesn't_ help make the set of strings immutable, as the caller can still add/remove elements from the set that was passed in (causing the state of the Builder's `alternativeNames` to change.

If the goal is to enforce immutability, then the set needs to be copied:
```java
this.alternativeNames.addAll(alternativeNames`)
```
and then make the copy immutable 
```java
this.alternativeNames = Collections.unmodifiableSet(builder.alternativeNames);
```
(typically it's preferable to leave the builder set mutable because there might be a need in the future to have setters for single elements that just _add_ to an existing set rather than complete replacement).

same for any other collection where immutability needs to be enforced.